### PR TITLE
vendor.mod: bump go line to 1.20

### PR DESCRIPTION
**Pull request by neersighted**
- Follow-up to/related to https://github.com/moby/moby/pull/46069
- Related to https://github.com/moby/moby/pull/45966#discussion_r1334862901

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like you included the other commits as well;  can you make it with only the `go.mod` changes?

**Comment by thaJeztah** [comment]
(given that we're already on go1.20, that should still work I think?)

File: .github/workflows/.windows.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ on:
         default: false
 
 env:
-  GO_VERSION: "1.20.8"
+  GO_VERSION: "1.21.1"
```
Comment (AkihiroSuda) [inline .github/workflows/.windows.yml:18]: Why demote?

Comment (thaJeztah) [reply .github/workflows/.windows.yml:18]: it's 1.20 -> 1.21 :) (not 1.20.8 -> 1.20.1)

Comment (thaJeztah) [reply .github/workflows/.windows.yml]: 😂 no worries! I saw your comment, and already thought you probably mis-read

**Comment by neersighted** [comment]
Ah, I intentionally stacked this on top of your branch, but you make a fair point that we can do this independent of your changes.

**Review by thaJeztah — APPROVED** [review]
LGTM

wondering if we still need the `-compat` option, or if we can remove that one (but that's perfectly fine for a separate PR); https://github.com/moby/moby/blob/a1d966c492c9f3941e1b6627f3f1128e76a27a2d/hack/vendor.sh#L13

**Comment by thaJeztah** [comment]
> Ah, I intentionally stacked this on top of your branch, but you make a fair point that we can do this independent of your changes.

Yes, and I think we should merge this PR before mine to have the changes in the desired order 😄

**Pull request merged**